pkg/chainaccessor/input_processors.go

```diff
@@ -0,0 +1,20 @@
+package chainaccessor
+
+import cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+
+// FilterOutChainSelector removes a specified chain selector from a slice of chain selectors
+func FilterOutChainSelector(
+	chains []cciptypes.ChainSelector,
+	chainToFilter cciptypes.ChainSelector) []cciptypes.ChainSelector {
+	if len(chains) == 0 {
+		return nil
+	}
+
+	filtered := make([]cciptypes.ChainSelector, 0, len(chains))
+	for _, chain := range chains {
+		if chain != chainToFilter {
+			filtered = append(filtered, chain)
+		}
+	}
+	return filtered
+}
```

Comment:
You can delete this function, there's already a generic helper function for it

```suggestion
func FilterOutChainSelector(
	chains []cciptypes.ChainSelector,
	chainToFilter cciptypes.ChainSelector
) []cciptypes.ChainSelector {
	return slicelib.Filter(chains, func(cs cciptype.ChainSelector) bool {
		return cs != chainToFilter
	}))
}
```